src/main/c/Makefile

```diff
@@ -24,6 +24,7 @@ libbwa.$(LIB_EXT): $(JNI_BASE_NAME).o jnibwa.o bwa/libbwa.a
 
 bwa:
 	git clone https://github.com/lh3/bwa && cd bwa && git checkout $(BWA_MEM_COMMIT) && echo '#define BWA_COMMIT "'$(BWA_MEM_COMMIT)'"' > bwa_commit.h
+	sed -i .bak 's/\(LOBJS=.*\)/\1 $$(AOBJS)/g' bwa/Makefile
```

Comment:
How about going down one level:  Other than the command line parsing there are only a dozen or so lines of code in bwa_index.  If you copied that code you could avoid non-library functions and avoid command-line parsing, too.